src/sentry/tasks/summaries/daily_summary.py

```diff
@@ -207,12 +207,23 @@ def build_summary_data(
                 group__in=([group for group in regressed_or_escalated_groups]),
                 type__in=(ActivityType.SET_REGRESSION.value, ActivityType.SET_ESCALATING.value),
             )
-            if regressed_or_escalated_groups_today:
-                for activity in regressed_or_escalated_groups_today[:4]:
-                    if activity.type == ActivityType.SET_REGRESSION.value:
-                        project_ctx.regressed_today.append(activity.group)
+
+            deduped_groups_by_activity_type = {"regressed": set(), "escalated": set()}
+            for activity in regressed_or_escalated_groups_today:
+                if activity.type == ActivityType.SET_REGRESSION.value:
+                    deduped_groups_by_activity_type["regressed"].add(activity.group)
+                else:
+                    if activity.type == ActivityType.SET_ESCALATING.value:
```

Comment:
This doesn't cover the case where a group has both regressed and escalated in the last 24 hours, I can do that as well but I wasn't sure if it was a realistic edge case.